script/VaultsDeployer.s.sol

```diff
@@ -0,0 +1,147 @@
+            )
+        );
+
+        syncManager = SyncManager(
+            create3(generateSalt("syncManager"), abi.encodePacked(type(SyncManager).creationCode, abi.encode(deployer)))
+        );
+
+        vaultRouter = VaultRouter(
+            create3(
+                generateSalt("vaultRouter"),
+                abi.encodePacked(
+                    type(VaultRouter).creationCode, abi.encode(address(routerEscrow), gateway, spoke, deployer)
+                )
+            )
+        );
+
+        asyncVaultFactory = AsyncVaultFactory(
+            create3(
+                generateSalt("asyncVaultFactory"),
+                abi.encodePacked(
+                    type(AsyncVaultFactory).creationCode, abi.encode(address(root), asyncRequestManager, deployer)
+                )
+            )
+        );
+
+        syncDepositVaultFactory = SyncDepositVaultFactory(
+            create3(
+                generateSalt("syncDepositVaultFactory"),
+                abi.encodePacked(
+                    type(SyncDepositVaultFactory).creationCode,
+                    abi.encode(address(root), syncManager, asyncRequestManager, deployer)
+                )
+            )
+        );
+
+        _vaultsRegister();
+        _vaultsEndorse();
+        _vaultsRely();
+        _vaultsFile();
+    }
+
+    function _vaultsRegister() private {
+        register("routerEscrow", address(routerEscrow));
+        register("globalEscrow", address(globalEscrow));
+        register("asyncRequestManager", address(asyncRequestManager));
+        register("syncManager", address(syncManager));
+        register("asyncVaultFactory", address(asyncVaultFactory));
+        register("syncDepositVaultFactory", address(syncDepositVaultFactory));
+        register("vaultRouter", address(vaultRouter));
+    }
+
+    function _vaultsEndorse() private {
+        root.endorse(address(asyncRequestManager));
+        root.endorse(address(globalEscrow));
+        root.endorse(address(vaultRouter));
+    }
+
+    function _vaultsRely() private {
+        // Rely Spoke
+        asyncVaultFactory.rely(address(spoke));
+        syncDepositVaultFactory.rely(address(spoke));
+        asyncRequestManager.rely(address(spoke));
+        syncManager.rely(address(spoke));
+
+        // Rely async requests manager
+        globalEscrow.rely(address(asyncRequestManager));
+
+        // Rely Root
+        vaultRouter.rely(address(root));
+        asyncRequestManager.rely(address(root));
+        syncManager.rely(address(root));
+        routerEscrow.rely(address(root));
+        globalEscrow.rely(address(root));
+        asyncVaultFactory.rely(address(root));
+        syncDepositVaultFactory.rely(address(root));
+
+        // Rely others
+        routerEscrow.rely(address(vaultRouter));
+        syncManager.rely(address(syncDepositVaultFactory));
+
+        // Rely VaultRouter
+        gateway.rely(address(vaultRouter));
```

Comment:
FYI. This rely dependency seems off since vaults are a layer on top of spoke now. But not sure how to fix it without opening any security issues in the Gateway.